src/_pytest/_io/terminalwriter.py

```diff
@@ -37,32 +41,39 @@ def should_do_markup(file: TextIO) -> bool:
     )
 
 
+def _ansi_items(inst, transform: Callable[[str], str] = str.lower) -> Dict[str, str]:
+    return {transform(name): cast(str, item) for name, item in vars(inst).items()}
+
+
+ESC_TABLE = {
+    **_ansi_items(ansi.Fore),
+    **_ansi_items(ansi.Back, str.capitalize),
+    **_ansi_items(ansi.Style),
+    # wrongly named in pylib
+    "purple": ansi.Fore.MAGENTA,
+    "Purple": ansi.Back.MAGENTA,
+    # missing, but hopefully unused
+    "invert": ansi.code_to_chars(7),
+    "blink": ansi.code_to_chars(5),
+    # wrongly named in pylib
+    "light": ansi.Style.DIM,
+    "bold": ansi.Style.BRIGHT,
+}
+RESET = ansi.Style.RESET_ALL
+
+
 @final
 class TerminalWriter:
-    _esctable = dict(
-        black=30,
-        red=31,
-        green=32,
-        yellow=33,
-        blue=34,
-        purple=35,
-        cyan=36,
-        white=37,
-        Black=40,
-        Red=41,
-        Green=42,
-        Yellow=43,
-        Blue=44,
-        Purple=45,
-        Cyan=46,
-        White=47,
-        bold=1,
-        light=2,
-        blink=5,
-        invert=7,
-    )
+    hasmarkup: bool
+    code_highlight: bool
 
-    def __init__(self, file: Optional[TextIO] = None) -> None:
+    def __init__(
+        self,
+        file: Optional[TextIO] = None,
+        dup=False,
```

Comment:
```suggestion
        dup: bool=False,
```

Actually it doesn't appear to be used?

Reply:
@nicoddemus whops, i  intend to add a dup, as we need to have a extra safe fd for printing when capture is active, i should have stashed it (or added it)